src/module-elasticsuite-core/Index/Indices/Config.php

```diff
@@ -154,6 +145,40 @@ private function initIndexConfig(array $indexConfigData)
         return ['types' => $types, 'defaultSearchType' => $defaultSearchType];
     }
 
+    /**
+     * Prepare mapping fields by merging static fields with dynamic ones.
+     *
+     * @param array $typeConfigData Processed type configuration.
+     * @param array $dataSources    Data sources for current type.
+     *
+     * @return \Smile\ElasticsuiteCore\Api\Index\Mapping\FieldInterface[]
+     */
+    private function getMappingFields($typeConfigData, $dataSources)
+    {
+        /** @var DynamicFieldProviderInterface[] $dynamicFieldProviders */
+        $dynamicFieldProviders = array_filter($dataSources, [$this, 'isDynamicFieldsProvider']);
+
+        $fields = [];
+
+        foreach ($dynamicFieldProviders as $dynamicFieldProvider) {
+            $fields += $dynamicFieldProvider->getFields();
+        }
+
+        foreach ($typeConfigData['mapping']['staticFields'] as $fieldName => $fieldConfig) {
+            if (isset($fields[$fieldName])) {
```

Comment:
Use mergeConfig method instead of this one :

```php
$field = $this->mappingFieldFactory->create(['name' => $fieldName] + $fieldConfig);

if (isset($fields[$fieldName]) {
    $field = $fields[$fieldName]->mergeConfig($fieldConfig);
}

$fields[$fieldName] = $field;
```